core/src/err.rs

```diff
@@ -54,6 +54,34 @@ pub enum Error {
     UnsupportedOperation(&'static str),
     #[error("Statement not executed")]
     StatementNotExecuted,
+    #[error(
+        "The ODBC driver version `{0}` is incompatible with libmongosqltranslate version `{1}`"
+    )]
+    LibmongosqltranslateLibraryIsIncompatible(&'static str, String),
+    #[error("The following collections were not found in the `__sql_schemas` collection: {0:?}")]
+    SchemaDocumentNotFoundInSchemaCollection(Vec<String>),
+    #[error(
+        "The libmongosqltranslate command `{0}` failed. Error message: `{1}`. Error is internal: {2}"
+    )]
+    LibmongosqltranslateCommandFailed(&'static str, String, bool),
+    #[error("Loading the runCommand symbol from libmongosqltranslate failed with error: {0}")]
+    RunCommandSymbolNotFound(String),
+    #[error("Deserializing libmongosqltranslate response to BSON Document failed with error: {0}")]
+    LibmongosqltranslateDeserialization(mongodb::bson::de::Error),
+    #[error("Deserializing libmongosqltranslate response Document to CommandResponse failed with error: {0}")]
+    BsonDocumentToCommandResponseDeserialization(mongodb::bson::de::Error),
```

Comment:
> Could we add a new error type for malformed responses and convert to it?

@terakilobyte Is this the new error you wanted added? It's for the conversion from `Document` to `CommandResponse`, so it will catch any incorrectly formatted `Documents`.